lib/flutter/flutter_tools.dart

```diff
@@ -78,19 +83,35 @@ class FlutterToolsManager implements Disposable {
       return;
     }
 
-    DartProject project = projectManager.getProjectFor(editor?.getPath());
+    DartProject project = projectManager.getProjectFor(editor.getPath());
     if (project == null) {
       atom.notifications.addWarning('The current project is not a Dart project.');
       return;
     }
 
     FlutterTool flutter = _flutterSdk.sdk.flutterTool;
-    flutter.runInJob(
-      ['upgrade'],
+    flutter.runInJob(['upgrade'],
       title: 'Running Flutter upgrade…',
       cwd: project.directory.path
     );
   }
 
+  void _doctor(AtomEvent _) {
+    if (!_flutterSdk.hasSdk) {
+      _flutterSdk.showInstallationInfo();
+      return;
+    }
+
+    FlutterTool flutter = _flutterSdk.sdk.flutterTool;
+
+    TextEditor editor = atom.workspace.getActiveTextEditor();
+    DartProject project = projectManager.getProjectFor(editor?.getPath());
+
+    flutter.runInJob(['doctor'],
+      title: 'Running Flutter doctor…',
+      cwd: project?.directory?.path
```

Comment:
Yeah, right now it's not expecting to be in a project dir. It's just testing to see what's installed on the platform (xcode, adb, ...).

I add this here more defensively in case `doctor` does start checking the local project dir.

Perhaps I should instead always have the cwd be the flutter root? Then it'd always be running from the same dir, instead of whatever the cwd of atom happened to be at the time.